portfolio/src/main/webapp/comments.js

```diff
@@ -70,11 +72,46 @@ class Comments {
     .then(this.getComments(1))
     .then(this.checkDeleteError)
     .catch((error) => {
-      console.log(error);
+      console.log(`${error}. Error occurred after attempting to delete comments.`);
     });
   }
 
   /**
+   * Retrieves login status and associated URL.
+   */
+  getLoginStatus() {
+    fetch('/authenticate-user')
+    .then(this.checkFetchError)
+    .then((loginStatus) => {
+      // Show comments if you're logged in.
+      if (loginStatus.isLoggedIn) {
+        this.getComments(10);
+      } 
+      // Always create a login-logout-button regardless of loginStatus.
+      this.createLoginLogoutButton(loginStatus.isLoggedIn, loginStatus.loginLogoutURL);
+    })
+    .catch((error) => {
+      console.log(`${error}. Failed to fetch login status of the user.`);
+    });
+  }
+
+  /**
+   * Create login-logout button depending on whether the user is logged in or not.
+   * @param {boolean} isLoggedIn - Is the user logged in? Yes or No?
+   * @param {string} loginLogoutURL - The URL the user will be
+   *   redirected once they click on the button.
+   */
+  createLoginLogoutButton(isLoggedIn, loginLogoutURL) {
+    if (isLoggedIn) {
+      this.loginLogoutButton.innerHTML = `<a href=${loginLogoutURL}>Log Out</a>`;
+      this.loginLogoutStatusMessage.innerText = 'Welcome! You are logged in!';
+    } else {
+      this.loginLogoutButton.innerHTML = `<a href=${loginLogoutURL}>Log In</a>`;
+      this.loginLogoutStatusMessage.innerText = 'Access Denied. Please log in!';
+    }
+  }
```

Comment:
Pretty sure this doesn't have anything to do with comments, right? This should be in its own class.